lib/jobs/run-sku-graph.js

```diff
@@ -76,17 +75,23 @@ function runSkuGraphJobFactory(
                 return;
             }
             if (!node.sku) {
+                // It's okay if there is no SKU, it just means there is
+                // nothing to do.
                 self._done();
                 return;
             }
 
             return waterline.skus.findOne({ id: node.sku })
             .then(function(sku) {
                 if (!sku) {
+                    self._done();
```

Comment:
@benbp , I agree the discovery graph should not fail if no SKU is specified, but we have handled the no-sku case in this line: https://github.com/RackHD/on-tasks/pull/157/files#diff-a4a2ba55165c416b61a5d04fe65e0dcaL78. For above line that I commented, the node has been assigned a valid SKU, if SKU document doesn't exist, so how the SKU was assigned before?